s4e-web/cypress/page-objects/auth/register.po.ts

```diff
@@ -0,0 +1,87 @@
+
+export interface User {
+  email: string;
+  name: string;
+  surname: string;
+  password: string;
+  repeatPassword: string;
+}
+
+export class Registration {
+  static readonly pageObject = {
+    getEmailInput: () => cy.get("#registration-login"),
+    getNameInput: () => cy.get("#registration-name"),
+    getSurnameInput: () => cy.get("#registration-surname"),
+    getPasswordInput: () => cy.get("#registration-password"),
+    getRepeatPasswordInput: () => cy.get("#registration-password-repeat"),
+    getReCaptachaCheckbox: () => cy.get(".recaptcha-checkbox-border"),
+    getSubmitBtn: () => cy.get('button[type="submit"]'),
+
+    getErrors: () => cy.get('.invalid-feedback > .ng-star-inserted'),
+
+  };
+
+  static fillForm(user: User) {
+    Registration.pageObject.getEmailInput().type(user.email);
+    Registration.pageObject.getNameInput().type(user.name);
+    Registration.pageObject.getSurnameInput().type(user.surname);
+    Registration.pageObject.getPasswordInput().type(user.password);
+    Registration.pageObject.getRepeatPasswordInput().type(user.repeatPassword);
+
+    return Registration;
+  }
+
+  static sendForm() {
+    Registration.pageObject.getSubmitBtn().click();
+
+    return Registration;
+  }
+
+  static errorsCountShouldBe(count: number) {
+    cy.location('pathname').should('eq', '/register');
+
+    Registration
+      .pageObject
+      .getErrors()
+      .should('have.length', count);
+
+    return Registration;
+  }
+
+  static clickReCaptcha() {
+
+    cy.server();
+    cy.route('POST', 'https://www.google.com/recaptcha/api2/reload*').as('captcha-reload')
+    cy.route('POST','https://www.google.com/recaptcha/api2/userverify*').as('captcha-userverify')
+
+    cy.get("iframe[src*=\"https://www.google.com/recaptcha\"]")
+      .then(($iframe) => {
+        const $body = $iframe.contents().find('body')
+
+        cy.wrap($body)
+          .find(".recaptcha-checkbox-border")
+
+          .click()
+      
+
+
+      })
+    cy.wait('@captcha-userverify')
+    cy.wait('@captcha-reload')
+
+
+    // cy.get("iframe[src*=\"https://www.google.com/recaptcha\"]")
```

Comment:
I would more likely use this block of code (commented one) instead of `cy.get("iframe[src*=\"https://www.google.com/recaptcha\"]").then`, because it's more reliable due to cypress recomendations.